### PR TITLE
Fix: calc max line and show context now work with folding

### PR DESCRIPTION
**Pull request by binhtddev**
This is a part from #3. This also add function to calculate relative line with folding but I haven't implemented it for line number.

Before:
old

After:
new

### CODE REVIEW

File: lua/treesitter-context.lua [hunk 1]
```diff
@@ -536,13 +540,31 @@ local function open(ctx_nodes)
...
+  local relativeline = 0
+  local i = line
+  while i < higher_line do
+    i = get_next_line(i)
+    relativeline = relativeline + 1
+  end
+  return relativeline
+end
+
 local function calc_max_lines(config_max)
```
Comment (binhtddev) [inline lua/treesitter-context.lua]: @lewis6991, Do you think we should remove this function? I just notice that I have added more condition to make sure context not overflow cursor without calc max line.

**Comment by lewis6991** [comment]
I'm really struggling to figure out what this PR actually does.

File: lua/treesitter-context.lua [hunk 2]
```diff
@@ -365,7 +365,7 @@ local function get_parent_matches(max_lines)
     end
   end
 
-  return reverse_table(parent_matches)
```
Comment (lewis6991) [inline lua/treesitter-context.lua:371]: Remove the `reverse_table` function in this commit, not the next.

File: lua/treesitter-context.lua [hunk 3]
```diff
@@ -307,16 +307,16 @@ local M = {
...
-    end
+--- Get next line with folding calculation
+---@param line string Current line
+---@return number _ Next line
+local function get_next_line(line)
+  local foldend_line = vim.fn.foldclosedend(line)
+  if (foldend_line ~= -1) then
+    return foldend_line + 1
+  else
+    return line + 1
```
Comment (lewis6991) [inline lua/treesitter-context.lua:318]: Introduce this function into a commit that uses it.

**Comment by binhtddev** [comment]
> I'm really struggling to figure out what this PR actually does.

Calc max lines is wrong when fold is between topline and cursor.
before

**Comment by binhtddev** [comment]
@lewis6991, I have just found max_lines is not working as before, If I set it to 1, it only show first line instead near cursor line.

**Comment by binhtddev** [comment]
I will close this PR because the current algorithm to calculate hidden context make max lines not work as expect. Thanks for replied!

**Pull request closed without merging**